
syntax = "proto3";

package AudioService.auqa.pb;

service AudioService {
  rpc Connect (ConnectRequest) returns (ConnectResponse) {}
  rpc Disconnect (DisconnectRequest) returns (DisconnectResponse) {}
  rpc KeepAlive (KeepAliveRequest) returns (KeepAliveResponse) {}
}

message ConnectRequest {
  string client_id = 1;
  string client_address = 2;
  uint32 client_port = 3;
}

message ConnectResponse {
  bool success = 1;
  string error_message = 2;
  string server_address = 3;
  uint32 server_port = 4;
}

message DisconnectRequest {
  string client_id = 1;
}

message DisconnectResponse {
  bool success = 1;
}

message KeepAliveRequest {
  string client_id = 1;
}

message KeepAliveResponse {
  bool success = 1;
}
